warehouse/query-core/src/main/java/datawave/query/iterator/ivarator/IvaratorCacheDirConfig.java

```diff
@@ -0,0 +1,138 @@
+package datawave.query.iterator.ivarator;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import org.apache.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class IvaratorCacheDirConfig {
+    private static final Logger log = Logger.getLogger(IvaratorCacheDir.class);
+    
+    public static final int DEFAULT_PRIORITY = Integer.MAX_VALUE;
+    public static final long DEFAULT_MIN_AVAILABLE_STORAGE_MB = 0L;
+    public static final double DEFAULT_MIN_AVAILABLE_STORAGE_PERCENT = 0f;
+    
+    // the base path for caching ivarator output for this filesystem
+    final protected String basePathURI;
+    
+    // a number >= 0 used to determine the order in which ivarator cache dirs are used (ascending order)
+    final protected int priority;
+    
+    // the minimum amount of available storage required to use this filesystem
+    final protected long minAvailableStorageMB;
+    
+    // the minimum percent of available storage required to use this filesystem
+    final protected double minAvailableStoragePercent;
+    
+    protected IvaratorCacheDirConfig() {
+        this(null);
+    }
+    
+    public IvaratorCacheDirConfig(String basePathURI) {
+        this(basePathURI, DEFAULT_PRIORITY);
+    }
+    
+    public IvaratorCacheDirConfig(String basePathURI, int priority) {
+        this(basePathURI, priority, DEFAULT_MIN_AVAILABLE_STORAGE_MB, DEFAULT_MIN_AVAILABLE_STORAGE_PERCENT);
+    }
+    
+    public IvaratorCacheDirConfig(String basePathURI, int priority, long minAvailableStorageMB) {
+        this(basePathURI, priority, minAvailableStorageMB, DEFAULT_MIN_AVAILABLE_STORAGE_PERCENT);
+    }
+    
+    public IvaratorCacheDirConfig(String basePathURI, int priority, double minAvailableStoragePercent) {
+        this(basePathURI, priority, DEFAULT_MIN_AVAILABLE_STORAGE_MB, minAvailableStoragePercent);
+    }
+    
+    private IvaratorCacheDirConfig(String basePathURI, int priority, long minAvailableStorageMB, double minAvailableStoragePercent) {
+        this.basePathURI = basePathURI;
+        this.priority = priority;
+        this.minAvailableStorageMB = minAvailableStorageMB;
+        this.minAvailableStoragePercent = minAvailableStoragePercent;
+    }
+    
+    @JsonIgnore
+    public boolean isValid() {
+        boolean result = true;
+        
+        if (!basePathURI.startsWith("file:/") && !basePathURI.startsWith("hdfs:/")) {
```

Comment:
Null check on basePathURI first, since it's null by default?

Reply:
Agreed.

Reply:
fixed in latest commit.